x/rollapp/keeper/hard_fork.go

```diff
@@ -195,3 +196,10 @@ func (k Keeper) pruneFinalizationsAbove(ctx sdk.Context, rollappID string, lastS
 	}
 	return nil
 }
+
+// is <height,revision> the first of the latest fork?
+func (k Keeper) ResolvesHardFork(ctx sdk.Context, rollappId string, revision, height uint64) bool {
+	rollapp := k.MustGetRollapp(ctx, rollappId)
+	latest := rollapp.LatestRevision().Number
+	return rollapp.DidFork() && rollapp.IsRevisionStartHeight(revision, height) && revision == latest
```

Comment:
Well this was exactly the original bug because of the hooks
The hook processing the state update caused another fork, so revision was not equal latest